prime-router/src/main/resources/db/migration/V3__new_lineage_tables.sql

```diff
@@ -0,0 +1,73 @@
+/*
+The Flyway tool applies this migration to create the database.
+
+Follow this style guide https://about.gitlab.com/handbook/business-ops/data-team/platform/sql-style-guide/
+use VARCHAR(63) for names in organization and schema
+
+Copy a version of this comment into the next migration
+
+*/
+/* 
+   This SQL creates tables to track lineage. 
+*/
+
+
+/*
+A task tracks a reports progress though the work pipeline. It also contains
+metadata about the body of the report. Each report in the system has a task.
+*/
+
+/*
+   Add a new action 'receive', and fix the current task table to use it.
+ */
+ALTER TABLE task ALTER COLUMN next_action TYPE VARCHAR(255);
+DROP TYPE IF EXISTS task_action;
+CREATE TYPE task_action AS ENUM ('receive', 'translate', 'batch', 'send', 'wipe', 'batch_error', 'send_error', 'wipe_error', 'none');
+ALTER TABLE task ALTER COLUMN next_action TYPE task_action USING next_action::task_action;
+
+-- Each row is an action already taken.
+CREATE TABLE action (
+    action_id SERIAL PRIMARY KEY,
+    action_name TASK_ACTION,
+    action_params VARCHAR(512),
+    action_result VARCHAR(2048),
+    -- Every table must have created_at timestamp
+    created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW()
+);
+
+-- Each row is a report, created by some action
+CREATE TABLE report_file (
+    report_id UUID PRIMARY KEY,
+    action_id INT NOT NULL REFERENCES action(action_id) ON DELETE CASCADE,
+
+    next_action TASK_ACTION,   -- what if there are multiple next actions?
+    next_action_at TIMESTAMP WITH TIME ZONE,
+
+    -- These are non-null only for 'receive' actions
+    sending_org VARCHAR(63),        -- should be a ref to an org table someday
+    sending_org_client VARCHAR(63), -- OrganizationClient
+
+    -- These are non-null only for 'send' actions:
+    receiving_org VARCHAR(63),      -- should be a ref to an org table someday
+    receiving_org_svc VARCHAR(63),  -- OrganizationService
+
```

Comment:
Note:  I took out the JSON `result` and `parameters` columns  from this table - they are now in the ACTION table.    I waffled over this for a long time, as they properly normalize to the ACTION.  It was only a convenience to have them in REPORT_FILE, and it was very redundant.  This is better, even though it will cause some extra sql joins.